src/extension/migration.js

```diff
@@ -0,0 +1,41 @@
+export default function migrateStyle(properties) {
+  const dataTemplateLabel = {
+    label: {
+      value: {
+        colorType: undefined,
+        color: undefined,
+        colorExpression: undefined,
+      },
+    },
+  };
+
+  const dataTemplateBackground = {
+    backgroundColor: {
+      colorType: undefined,
+      color: undefined,
+      colorExpression: undefined,
+    },
+  };
+
+  const dataTemplateBorder = {
+    border: {
+      top: undefined,
+      fullBorder: undefined,
+      colorType: undefined,
+      color: undefined,
+      colorExpression: undefined,
+    },
+  };
+
+  if (!properties.components || properties.components.length === 0) {
```

Comment:
What would happen when you reset the styling panel? That removes all the entries in the components array I think. Do you migrate again?

Reply:
No, it will not migrate again. The reset will clean up the old styling and use the default values for the properties instead.

Reply:
This PR only adds the styling panel and applies the migrated values. All the logic to apply the selections made on the styling panel will be included ini a second PR.

Reply:
Okay, that might give issues with backwards compatibility I guess. But sometimes that is not something we can go without.

Reply:
Yes, that's true. The styling defined with the styling panel will not be applied if we turn off the feature flag afterwards. PM approved with that caveat.